csrc/executor.cpp

```diff
@@ -1342,34 +1342,25 @@ std::vector<FusionExecutor::GlobalBufferInfo> FusionExecutor::
   return global_buffers;
 }
 
-std::vector<at::Tensor> FusionExecutor::allocOutputSpace(
-    const at::ArrayRef<c10::IValue>& inputs) {
-  auto kernel_inputs = KernelArgumentHolder::createKernelArgumentHolder(inputs);
-  auto expr_eval =
-      executor_utils::bindInputs(kernel_inputs, lowered_->kernel());
-
-  auto output_info =
-      getOutputBufferInfo(kernel_inputs, expr_eval, kernel()->indexType());
-
-  return allocateOutputs(kernel(), output_info, options_.device, expr_eval);
-}
-
-std::vector<FusionExecutor::GlobalBufferInfo> FusionExecutor::
-    getOutputBufferInfo(
-        const KernelArgumentHolder& args,
-        ExpressionEvaluator& expr_eval,
-        DataType index_dtype) {
+//! Return information necessary for allocating output tensors. Input
+//! and output tensors are allowed to alias each other, which is
+//! specified by the list of int pairs of input and output indices
+std::vector<FusionExecutor::GlobalBufferInfo> getOutputBufferInfo(
+    const KernelArgumentHolder& args,
+    ExpressionEvaluator& expr_eval,
+    DataType index_dtype,
+    const kir::Kernel* kernel) {
   FUSER_PERF_SCOPE("FusionExecutor::getOutbufferInfo");
-  const auto kernel = lowered_->kernel();
-  std::vector<GlobalBufferInfo> outputs;
+  std::vector<FusionExecutor::GlobalBufferInfo> outputs;
+  outputs.reserve(kernel->outputs().size());
   NVF_ERROR(
       args.size() == kernel->inputs().size(),
       "kernel arguments length does not match runtime arguments.");
   for (const auto out_i : c10::irange(kernel->outputs().size())) {
     auto out_val = kernel->outputs()[out_i];
     auto output = out_val->as<TensorView>();
```

Comment:
Looks like we have `output`, which is a TensorView pointer, and also we do dynamic_cast with `out_val`. 

Maybe just delete `output` and replace it with `info.tv`.